api/scpca_portal/test/models/test_job.py

```diff
@@ -4,64 +4,102 @@
 from django.conf import settings
 from django.test import TestCase
 
+from botocore.exceptions import ClientError
+
 from scpca_portal.enums import JobStates
 from scpca_portal.models import Job
+from scpca_portal.test.factories import JobFactory
 
 
 class TestJob(TestCase):
     def setUp(self):
-        self.mock_job_id = "FAKE_JOB_ID"
-        self.project_id = "FAKE_PROJECT_ID"
-        self.download_config_name = "FAKE_DOWNLOAD_CONFIG"
-        self.batch_job_name = f"{self.project_id}-{self.download_config_name}"
-        self.project_job_batch_container_overrides = {
-            "command": [
-                "python",
-                "manage.py",
-                "generate_computed_file",
-                "--project-id",
-                self.project_id,
-                "--download-config-name",
-                self.download_config_name,
-                "",
-            ]
-        }
+        self.mock_project_id = "SCPCP000000"
+        self.mock_sample_id = "SCPCS000000"
+        self.mock_batch_job_id = "MOCK_JOB_ID"  # The job id via AWS Batch response
 
     @patch("scpca_portal.models.Job._batch")
     def test_submit_job(self, mock_batch_client):
-        # Set the mock return value of submit_job
-        mock_batch_client.submit_job.return_value = {"jobId": self.mock_job_id}
+        # Set up mock for submit_job
+        mock_batch_client.submit_job.return_value = {"jobId": self.mock_batch_job_id}
 
         job = Job.get_project_job(
-            project_id=self.project_id, download_config_name=self.download_config_name
+            project_id=self.mock_project_id, download_config_name="MOCK_DOWNLOAD_CONFIG"
         )
 
         # Before submission, the job instance should not have an ID
         self.assertIsNone(job.id)
 
         job.submit()
-
-        mock_batch_client.submit_job.assert_called_once_with(
-            jobName=self.batch_job_name,
-            jobQueue=settings.AWS_BATCH_JOB_QUEUE_NAME,
-            jobDefinition=settings.AWS_BATCH_JOB_DEFINITION_NAME,
-            containerOverrides=self.project_job_batch_container_overrides,
-        )
```

Comment:
Since we're checking the saved job instance state in the database, I've omitted the check before saving the instance, as they are identical. I've applied the same logic to the tests for `Job::terminate` as well.